FitGoal/FitGoal/Singup&Singin/CustomUIViews/AuthenticationFormView.swift

```diff
@@ -63,6 +69,31 @@ class AuthenticationFormView: UIStackView, UITextFieldDelegate {
         return true
     }
     
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        if let textField = textField as? CustomTextField {
+            if !textField.textFieldMessageLabel.isHidden {
+                textField.textFieldMessageLabel.isHidden = true
+            }
+        }
+    }
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if let textField = textField as? CustomTextField {
+            guard let text = textField.text else { fatalError() }
+            delegate?.userDidEndEditing(textFieldType: textField.textFieldType, with: text)
+        }
+    }
+    
+    func showAuthenticationMessage(message: String, in textField: TextFieldType) {
+        for customTextField in self.subviews {
+            if let field = customTextField as? CustomTextField, textField == field.textFieldType {
+                field.textFieldMessageLabel.isHidden = false
+                field.formatPlaceholderLabel(with: message)
```

Comment:
Same as the comment above talking about the abstraction of the custom text field.